rotkehlchen/assets/spam_assets.py

```diff
@@ -1039,24 +1041,43 @@
 ]
 
 
-def query_token_spam_list(db: 'DBHandler') -> set[EvmToken]:
+def query_token_spam_list() -> set[Asset]:
     """
     Generate a set of assets that can be ignored using the list of spam
     assets KNOWN_EVM_SPAM_TOKENS.
     """
     tokens_to_ignore = set()
     # Try to add custom list
     for info in KNOWN_EVM_SPAM_TOKENS:
-        own_token = get_or_create_evm_token(
-            userdb=db,
-            evm_address=info['address'],
+        evm_token = EvmToken.initialize(
+            address=info['address'],
             chain_id=info.get('chain', ChainID.ETHEREUM),
-            protocol=SPAM_PROTOCOL,
-            decimals=info.get('decimals', 18),
+            token_kind=EvmTokenKind.ERC20,
             name=info.get('name', MISSING_NAME_SPAM_TOKEN),
+            decimals=info.get('decimals', 18),
             symbol=info.get('symbol', MISSING_SYMBOL_SPAM_TOKEN),
+            protocol=SPAM_PROTOCOL,
+            underlying_tokens=None,
         )
-        tokens_to_ignore.add(own_token)
+        token_exists = True
+        try:
+            evm_token.check_existence()
+        except UnknownAsset:
+            token_exists = False
+
+        if token_exists is True:
+            # make sure that the token has the spam protocol
+            db_evm_token = EvmToken(evm_token.identifier)
+            if db_evm_token.protocol != SPAM_PROTOCOL:
+                GlobalDBHandler().edit_evm_token(entry=evm_token)
+        else:
+            GlobalDBHandler().add_asset(
+                asset_id=evm_token.identifier,
+                asset_type=AssetType.EVM_TOKEN,
+                data=evm_token,
+            )
+        # save the asset instead of the EvmToken as we don't need all the extra information later
+        tokens_to_ignore.add(Asset(evm_token.identifier))
```

Comment:
shorter and no need to use extra variable

```suggestion
        try:
            evm_token.check_existence()
        except UnknownAsset:  # token does not exist
            GlobalDBHandler().add_asset(
                asset_id=evm_token.identifier,
                asset_type=AssetType.EVM_TOKEN,
                data=evm_token,
            )
        else:  # token exists, make sure it has spam protocol set
            db_evm_token = EvmToken(evm_token.identifier)
            if db_evm_token.protocol != SPAM_PROTOCOL:
                GlobalDBHandler().edit_evm_token(entry=evm_token)
        # save the asset instead of the EvmToken as we don't need all the extra information later
        tokens_to_ignore.add(Asset(evm_token.identifier))
```

Reply:
I wanted to avoid the error to control the flow but okey, will change

Reply:
there is no change in the flow. Same thing happens.